healthcheck.go

```diff
@@ -53,6 +55,71 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if an index from the client exists
+//What is a context?
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	var clientElasticSearchIndices = cli.indices
+
+	for _, index := range clientElasticSearchIndices {
+
+		urlIndex := cli.url + "/" + index
+		logData := log.Data{"url": urlIndex}
+
+		URL, err := url.Parse(urlIndex)
+		if err != nil {
+			log.Event(ctx, "failed to create url for elasticsearch indexcheck", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		path := URL.String()
+		//What do the ["url"] brackets mean - is it resetting the value for the "url" key to the path
+		logData["url"] = path
+
+		//what is background context?
+		req, err := http.NewRequest("HEAD", path, nil)
+		if err != nil {
+			log.Event(ctx, "failed to create request for indexcheck call to elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		//Where do the credentials come from?
+		if cli.signRequests {
+			awsauth.Sign(req)
+		}
+
+		resp, err := cli.httpCli.Do(ctx, req)
+		if err != nil {
+			log.Event(ctx, "failed to call elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+		defer resp.Body.Close()
+		logData["http_code"] = resp.StatusCode
+
+		switch resp.StatusCode {
+
+		case 200:
+			return resp.StatusCode, nil
+		case 404:
+			log.Event(ctx, "404 status code returned in response", logData, log.ERROR, log.Error(ErrorIndexDoesNotExist))
+			return resp.StatusCode, ErrorIndexDoesNotExist
+		default:
+			log.Event(ctx, "unexpected status code returned in response", logData, log.ERROR, log.Error(ErrorUnexpectedStatusCode))
+			return resp.StatusCode, ErrorUnexpectedStatusCode
+		}
+
+	}
+
+	//I think I need to return something here (there is an error if not) - just put in this to see if it got rid of the error
+	//Would it return here if it was unable to form the var clientElasticSearchIndices = cli.indices because it couldn't find the cli.indices?
```

Comment:
I imagine cli.indices exists at this point. Maybe you can check before

Reply:
Maybe I will add the check before the range statement

Reply:
I think the indexes array will always exist (as required in client Constructors, and it's not a pointer, so i can't be nil).